src/components/Checkbox/view/alpha/style/Checkbox.style.ts

```diff
@@ -0,0 +1,188 @@
+import styled, { css } from 'styled-components';
+import { Size } from '../../../../../types';
+import { StyledIcon } from '../../../../Icon';
+import { getDefaultTypographyStyles } from '../../../../Typography/style';
+
+type StyledCheckboxProps = {
+    isChecked?: boolean;
+    isPartiallyChecked?: boolean;
+    isDisabled?: boolean;
+    isTextDotted?: boolean;
+    name?: string;
+    position?: 'right' | 'left';
+    size?: Size;
+};
+
+export const HiddenCheckbox = styled.input.attrs({
+    type: 'checkbox',
+})`
+    opacity: 0;
+    width: 0;
+    height: 0;
+    margin: 0;
+    position: absolute;
+    z-index: -1;
+`;
+
+export const StyledCheckboxInputShown = styled('div').withConfig({
+    shouldForwardProp: (prop) => !['isDisabled'].includes(prop),
+})<StyledCheckboxProps>`
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    padding: ${(props) => parseInt(props.theme.padding.xs, 0) - 1}px;
+    background: ${(props) => props.theme.color.background.alpha};
+    border-radius: ${(props) => parseInt(props.theme.radius, 0) - 1}px;
+    border: 1px solid ${(props) => props.theme.color.border};
+
+    ${(props) =>
+        props.isDisabled &&
+        css`
+            background-color: ${props.theme.color.default};
+            cursor: not-allowed;
+        `}
+`;
+
+export const StyledCheckboxLabel = styled.label.withConfig({
+    shouldForwardProp: (prop) => !['isDisabled', 'size'].includes(prop),
+})<StyledCheckboxProps>`
+    ${(props) => getDefaultTypographyStyles(props)};
+    display: inline-flex;
+    align-items: center;
+    height: ${(props) => props.theme.height.sm};
```

Comment:
Pokud zajistíš, aby komponenta dostala nějakou výchozí velikost tak zde to psát nemusíš. Stačí když máš ty podmínky na velikosti níže.

Reply:
ok